src/backend/aspen/api/views/auth.py

```diff
@@ -0,0 +1,68 @@
+import os
+from urllib.parse import urlencode
+
+from authlib.integrations.base_client.errors import OAuthError
+from authlib.integrations.starlette_client import OAuth
+from fastapi import APIRouter
+from fastapi.responses import RedirectResponse
+from starlette.requests import Request
+
+import aspen.api.error.http_exceptions as ex
+from aspen.api.config.config import settings
+
+# From the example here:
+# https://github.com/authlib/demo-oauth-client/tree/master/fastapi-google-login
+router = APIRouter()
+
+oauth = OAuth()
+auth0 = oauth.register(
+    "auth0",
+    client_id=settings.AUTH0_CLIENT_ID,
+    client_secret=settings.AUTH0_CLIENT_SECRET,
+    api_base_url=settings.AUTH0_BASE_URL,
+    access_token_url=settings.AUTH0_ACCESS_TOKEN_URL,
+    authorize_url=settings.AUTH0_AUTHORIZE_URL,
+    client_kwargs=settings.AUTH0_CLIENT_KWARGS,
+)
+
```

Comment:
These endpoints implement a basic oauth login flow, and are intended to be functionally identical to their counterparts in the flask app. This implementation borrows heavily from:
https://github.com/authlib/demo-oauth-client/tree/master/fastapi-google-login